### DISCUSSION TITLE
How to set "request.jwt.claims" ?

### DISCUSSION THREAD

**Discussion by sheldondz** [PostgREST/postgrest#2291] [Q&A] [answered]
Not a bug just a question

We have 2 apps running connected to the same database
1. postgrest app for read only apis to the database
2. custom apis for other business logic

We want both the apis to use the same row level security rules to grant api access to tables.

How do we pass the "request.jwt.claims" from the custom api so that the rules run successfully.

```
SET request.jwt.claims = some value
```
Is the above the correct way to do this?

**Comment by steve-chavez** [comment, marked as answer]
**Edit**: More specific:

It should be like:

```sql
 SELECT set_config('request.jwt.claims', '{"user_id": 1}', true);
```
---

> SET request.jwt.claims = some value

It should be a `SET LOCAL request.jwt.claims = some value`, for each request.

We actually do `select set_config('xx', 'yy', true)`, which is equivalent.

https://github.com/PostgREST/postgrest/blob/d6ec171bcbc7c095c250cc0d98ef60ec5ebbc225/src/PostgREST/Middleware.hs#L108-L112

**Reply by elimisteve** [reply]
@steve-chavez ^^This^^ link no longer works, plus `SET LOCAL request.jwt.claims = some value` doesn't tell us what format the value should be in. I tried `SELECT set_config('request.jwt.claims', '[{"user_id": 1}]', true);` but I don't know whether this matches what PostgREST internally does, so please be more specific, thanks.

**Reply by steve-chavez** [reply]
@elimisteve Sorry about that, corrected. The claims are set as a json object.

**Reply by elimisteve** [reply]
@steve-chavez Thank you! :+1:
